lua/nvim-web-devicons.lua

```diff
@@ -377,15 +381,31 @@ local function get_highlight_foreground(icon_data)
     icon_data = default_icon
   end
 
-  return string.format("#%06x", nvim_get_hl_by_name(get_highlight_name(icon_data), true).foreground)
+  local higroup = get_highlight_name(icon_data)
+
+  local fg
+  if vim.fn.has "nvim-0.9" == 1 then
+    fg = vim.api.nvim_get_hl(0, { name = higroup, link = false }).fg
+  else
+    fg = vim.api.nvim_get_hl_by_name(higroup, true).foreground ---@diagnostic disable-line: deprecated
+  end
+
+  return string.format("#%06x", fg)
 end
 
 local function get_highlight_ctermfg(icon_data)
   if not global_opts.color_icons then
     icon_data = default_icon
   end
 
-  return nvim_get_hl_by_name(get_highlight_name(icon_data), false).foreground
```

Comment:
`highlight_exists` is always called before this, hence the table is always returned.